autoscale-core/src/main/java/com/github/autoscaler/core/ScalerThread.java

```diff
@@ -252,46 +263,107 @@ public void scaleDownNow() throws ScalerException
+                resourceLimitStagesReached.getMemoryLimitStageReached(),
+                resourceLimitStagesReached.getDiskLimitStageReached());
+
+        if (highestResourceLimitStageReached == 1 && shutdownPriority <= resourceConfig.getResourceLimitOneShutdownThreshold()) {
+            LOG.debug("Attempting to scale down service {} due to resource limit stage 1 being reached and " +
+                            "shutdownPriority {} <= resourceLimitOneShutdownThreshold {}. The current resource limit stage is " +
+                            "determined by the highest resource limit stage among: {}",
+                    serviceRef,
+                    shutdownPriority,
+                    resourceConfig.getResourceLimitOneShutdownThreshold(),
+                    resourceLimitStagesReached);
 
-        if (currentMemoryLoadLimit == 1 && shutdownPriority <= resourceConfig.getResourceLimitOneShutdownThreshold()) {
             scaleDown(instances.getTotalRunningAndStageInstances());
             return true;
-        } else if (currentMemoryLoadLimit == 2 && shutdownPriority <= resourceConfig.getResourceLimitTwoShutdownThreshold()) {
+        } else if (highestResourceLimitStageReached == 2 && shutdownPriority <= resourceConfig.getResourceLimitTwoShutdownThreshold()) {
+            LOG.debug("Attempting to scale down service {} due to resource limit stage 2 being reached and " +
+                            "shutdownPriority {} <= resourceLimitTwoShutdownThreshold {}. The current resource limit stage is " +
+                            "determined by the highest resource limit stage among: {}",
+                    serviceRef,
+                    shutdownPriority,
+                    resourceConfig.getResourceLimitTwoShutdownThreshold(),
+                    resourceLimitStagesReached);
+
             scaleDown(instances.getTotalRunningAndStageInstances());
             return true;
-        } else if (currentMemoryLoadLimit == 3 && shutdownPriority <= resourceConfig.getResourceLimitThreeShutdownThreshold()) {
+        } else if (highestResourceLimitStageReached == 3 && shutdownPriority <= resourceConfig.getResourceLimitThreeShutdownThreshold()) {
+            LOG.debug("Attempting to scale down service {} due to resource limit stage 3 being reached and " +
+                            "shutdownPriority {} <= resourceLimitThreeShutdownThreshold {}. The current resource limit stage is " +
+                            "determined by the highest resource limit stage among: {}",
+                    serviceRef,
+                    shutdownPriority,
+                    resourceConfig.getResourceLimitThreeShutdownThreshold(),
+                    resourceLimitStagesReached);
+
             scaleDown(instances.getTotalRunningAndStageInstances());
             return true;
         }
         return false;
     }
 
-    private void handleAlerterDispatch(final double memLoad) throws ScalerException
```

Comment:
There were 2 bugs in this existing method:

1) `memLoad` param was actually the stage (1, 2 or 3), when it should have been the current % used
2) It was using `>` rather than `>=` (which is what is used in the establishMemLimitReached method below) to compare against the threshold

Point (1) meant that we never sent out alerts.

Point (2)  meant that we were scaling down workers when they reached e.g. stage 1 (e.g. 70%), but not sending an email alert until the memory load was > 70%, which did not seem correct, so I have changed this to `>=` in the updated `handleAlerterDispatch` method.